Stub out required functions in redis_cache

carsonoid opened the pull request:
Without these functions the masters are
not able to use the redis cache completely

Signed-off-by: Carson Anderson <ca@carsonoid.net>

### What does this PR do?

Adds some required stubs.

### What issues does this PR fix or reference?

N/A

### Previous Behavior

Masters could not use redis to get pillar data or check the responses from minions.

For example,. given a salt installation with two minions that are up and one that is down this is the result of a `test.ping`:

```
root@ip-172-30-6-82:~# salt \* test.ping
ip-172-30-7-18.ec2.internal:
    True
ip-172-30-8-18.ec2.internal:
    True
[ERROR   ] Exception raised when processing __virtual__ function for salt.loaded.int.cache.consul. Module will not be loaded: 'module' object has no attribute 'Consul'
[WARNING ] salt.loaded.int.cache.consul.__virtual__() is wrongly returning `None`. It should either return `True`, `False` or a new name. If you're the developer of the module 'consul', please fix this.
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
KeyError: 'redis.ls'
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 476, in salt_main
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/salt.py", line 173, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.7/dist-packages/salt/client/__init__.py", line 809, in cmd_cli
    **kwargs):
  File "/usr/lib/python2.7/dist-packages/salt/client/__init__.py", line 1601, in get_cli_event_returns
    connected_minions = salt.utils.minions.CkMinions(self.opts).connected_ids()
  File "/usr/lib/python2.7/dist-packages/salt/utils/minions.py", line 577, in connected_ids
    search = self.cache.ls('minions')
  File "/usr/lib/python2.7/dist-packages/salt/cache/__init__.py", line 244, in ls
    return self.modules[fun](bank, **self._kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1113, in __getitem__
    func = super(LazyLoader, self).__getitem__(item)
  File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 101, in __getitem__
    raise KeyError(key)
KeyError: 'redis.ls'
Traceback (most recent call last):
  File "/usr/bin/salt", line 10, in <module>
    salt_main()
  File "/usr/lib/python2.7/dist-packages/salt/scripts.py", line 476, in salt_main
    client.run()
  File "/usr/lib/python2.7/dist-packages/salt/cli/salt.py", line 173, in run
    for full_ret in cmd_func(**kwargs):
  File "/usr/lib/python2.7/dist-packages/salt/client/__init__.py", line 809, in cmd_cli
    **kwargs):
  File "/usr/lib/python2.7/dist-packages/salt/client/__init__.py", line 1601, in get_cli_event_returns
    connected_minions = salt.utils.minions.CkMinions(self.opts).connected_ids()
  File "/usr/lib/python2.7/dist-packages/salt/utils/minions.py", line 577, in connected_ids
    search = self.cache.ls('minions')
  File "/usr/lib/python2.7/dist-packages/salt/cache/__init__.py", line 244, in ls
    return self.modules[fun](bank, **self._kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/loader.py", line 1113, in __getitem__
    func = super(LazyLoader, self).__getitem__(item)
  File "/usr/lib/python2.7/dist-packages/salt/utils/lazy.py", line 101, in __getitem__
    raise KeyError(key)
KeyError: 'redis.ls'

```

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

salt-jenkins commented:
@carsonoid, thanks for your PR! By analyzing the history of the files in this pull request, we identified @mirceaulinic and @rallytime to be potential reviewers.

gtmanfred commented:
@carsonoid fyi, since this is a bug fix, it should be opened against the oldest supported release that the bug is in.

https://docs.saltstack.com/en/latest/topics/development/contributing.html#the-current-release-branch

Which I believe for the redis_cache, it was added in 2017.7.0, so you should open it against the 2017.7 branch.

mirceaulinic commented on salt/cache/redis_cache.py:
```diff
@@ -493,6 +496,11 @@ def list_(bank):
         return []
     return list(bank_keys)
 
+def ls(bank):
```
To avoid redundant code, I'd recommend renaming the above function `list_` to `ls`, while the `__func_alias__` dunder should be `{'list': 'ls'}`.

mirceaulinic commented on salt/cache/redis_cache.py:
```diff
@@ -197,6 +197,9 @@ def __virtual__():
 # -----------------------------------------------------------------------------
 
 
+def init_kwargs(kwargs):
```
Where is that used?

carsonoid replied:
I'm not sure. I was just trying to clean up this log message I kept getting.

```
[DEBUG   ] Could not LazyLoad redis.init_kwargs: 'redis.init_kwargs' is not available.
```

Commit:
Stub out required functions for redis cache

Without these functions the masters are
not able to use the redis cache completely

Signed-off-by: Carson Anderson <ca@carsonoid.net>

carsonoid commented:
@gtmanfred updated PR base branch.

@mirceaulinic moved to __func_alias__

gtmanfred commented on salt/cache/redis_cache.py:
```diff
@@ -415,7 +417,7 @@ def flush(bank, key=None):
 
 
-def list_(bank):
+def list(bank):
```
you should make this `def ls`

mirceaulinic commented on salt/cache/redis_cache.py:
```diff
@@ -115,7 +115,7 @@
 __virtualname__ = 'redis'
 __func_alias__ = {
-    'list_': 'list'
+    'list': 'ls'
```
And reverse this key-value: `ls`: `list` (sorry about that, my bad)

carsonoid commented:
@gtmanfred  and @mirceaulinic changed as requested.

rallytime commented:
@carsonoid Can you fix up the lint error really quickly? 

https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/13610/violations/file/salt/cache/redis_cache.py/

carsonoid commented:
Fixed!

On Fri, Aug 11, 2017 at 12:07 PM, Nicole Thomas <notifications@github.com>
wrote:

> @carsonoid <https://github.com/carsonoid> Can you fix up the lint error
> really quickly?
>
> https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/
> 13610/violations/file/salt/cache/redis_cache.py/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/saltstack/salt/pull/42868#issuecomment-321881472>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AC1fSTKc7ZllAbTXP-oZzGa8yTSqK5Xkks5sXJhZgaJpZM4O0H1B>
> .
>

cachedout commented:
@mirceaulinic Could you please re-review this?

mirceaulinic reviewed (approved):
This looks good. Thanks @carsonoid
